staging/src/k8s.io/client-go/util/cert/server_inspection.go

```diff
@@ -43,9 +44,9 @@ func GetClientCANames(apiHost string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer conn.Close()
+	err = conn.Close()
```

Comment:
if we are going to care about the error (and doing so is ok), I think we should return an unusable value from the function.

```go
if err := conn.Close(); err != nil{
   return nil, err
}
```

Reply:
@deads2k 
Your comment has been addressed.